### DISCUSSION TITLE
Fail to compile NekRS

### DISCUSSION THREAD

**Discussion by aburbas** [Nek5000/nekRS#581] [Compiling]
Good afternoon Nek users,

I am trying to install and build NekRS on a cluster but experience problems with it on building step.
 
I believe I follow the instructions on the github page, but may be I missed something.
I would welcome any help and recommendations.

While building, I recieve several errors such as(I attach CMake error file):
C SOURCE FILE Test Allowed_LD_Flag_BSYMBOLIC failed

The modules I loaded:
 1) gcc/13.2.0 
 2) openmpi-gcc/openmpi5.0.3-gcc11.3.1 
 3) cmake/3.24.1

The command I use:

CC=mpicc CXX=mpic++ FC=mpif77 ./nrsconfig -DCMAKE_INSTALL_PREFIX=$HOME/.local/nekrs

Best,
Alexey
[CMakeError.log](https://github.com/user-attachments/files/16353655/CMakeError.log)
[CMakeOutput.log](https://github.com/user-attachments/files/16353656/CMakeOutput.log)

**Comment by aburbas** [comment]
Here is build attempt:
cmake -S . -B build -Wfatal-errors -DCMAKE_INSTALL_PREFIX=/home/aburbas/.local/nekrs
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- The Fortran compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpic++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpif77 - skipped
-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
-- Found MPI_C: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpicc (found version "3.1")
-- Found MPI_CXX: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpic++ (found version "3.1")
-- Found MPI_Fortran: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpif77 (found version "3.1")
-- Found MPI: TRUE (found version "3.1")
-- Verifying Fortran/CXX Compiler Compatibility
-- Verifying Fortran/CXX Compiler Compatibility - Success
-- Found MPI_UNDERLYING_COMPILER: /usr/local/apps/gcc/13.2.0/bin/g++
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP_Fortran: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Performing Test Allowed_LD_Flag_BSYMBOLIC
-- Performing Test Allowed_LD_Flag_BSYMBOLIC - Failed
-- Using CMake version 3.24.1
-- Performing Test Allowed_CXX_Flag_Wall__Wextra
-- Performing Test Allowed_CXX_Flag_Wall__Wextra - Success
-- Performing Test Allowed_CXX_Flag_Wunused_function__Wunused_variable
-- Performing Test Allowed_CXX_Flag_Wunused_function__Wunused_variable - Success
-- Performing Test Allowed_CXX_Flag_Wwrite_strings__Wfloat_equal
-- Performing Test Allowed_CXX_Flag_Wwrite_strings__Wfloat_equal - Success
-- Performing Test Allowed_CXX_Flag_Wcast_align__Wlogical_op
-- Performing Test Allowed_CXX_Flag_Wcast_align__Wlogical_op - Success
-- Performing Test Allowed_CXX_Flag_Wshadow
-- Performing Test Allowed_CXX_Flag_Wshadow - Success
-- Performing Test Allowed_CXX_Flag_Wno_unused_parameter
-- Performing Test Allowed_CXX_Flag_Wno_unused_parameter - Success
-- Performing Test Allowed_CXX_Flag_diag_disable_11074__diag_disable_11076
-- Performing Test Allowed_CXX_Flag_diag_disable_11074__diag_disable_11076 - Failed
-- Performing Test Allowed_CXX_Flag__display_error_number
-- Performing Test Allowed_CXX_Flag__display_error_number - Failed
-- Performing Test Allowed_CXX_Flag_Werror
-- Performing Test Allowed_CXX_Flag_Werror - Success
-- Performing Test Allowed_C_Flag_Wall__Wextra
-- Performing Test Allowed_C_Flag_Wall__Wextra - Success
-- Performing Test Allowed_C_Flag_Wunused_function__Wunused_variable
-- Performing Test Allowed_C_Flag_Wunused_function__Wunused_variable - Success
-- Performing Test Allowed_C_Flag_Wwrite_strings__Wfloat_equal
-- Performing Test Allowed_C_Flag_Wwrite_strings__Wfloat_equal - Success
-- Performing Test Allowed_C_Flag_Wcast_align__Wlogical_op
-- Performing Test Allowed_C_Flag_Wcast_align__Wlogical_op - Success
-- Performing Test Allowed_C_Flag_Wshadow
-- Performing Test Allowed_C_Flag_Wshadow - Success
-- Performing Test Allowed_C_Flag_Wno_c__11_long_long
-- Performing Test Allowed_C_Flag_Wno_c__11_long_long - Success
-- Performing Test Allowed_C_Flag_diag_disable_11074__diag_disable_11076
-- Performing Test Allowed_C_Flag_diag_disable_11074__diag_disable_11076 - Failed
-- Performing Test Allowed_C_Flag__display_error_number
-- Performing Test Allowed_C_Flag__display_error_number - Failed
-- Performing Test Allowed_C_Flag_Werror
-- Performing Test Allowed_C_Flag_Werror - Success
-- Performing Test COMPILER_SUPPORTS_NO_STRICT_ALIASING
-- Performing Test COMPILER_SUPPORTS_NO_STRICT_ALIASING - Success
-- System     : Linux-5.14.0-427.24.1.el9_4.x86_64
-- Build type : RelWithDebInfo
-- C flags    :  -Wall -Wextra -Wunused-function -Wunused-variable -Wwrite-strings -Wfloat-equal -Wcast-align -Wlogical-op -Wshadow -Wno-c++11-long-long  -fopenmp -O2 -g -march=native -mtune=native -ftree-vectorize
-- CXX flags  :  -Wall -Wextra -Wunused-function -Wunused-variable -Wwrite-strings -Wfloat-equal -Wcast-align -Wlogical-op -Wshadow -Wno-unused-parameter  -fopenmp -fno-strict-aliasing -O2 -g -march=native -mtune=native -ftree-vectorize
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP_Fortran: -fopenmp (found version "4.5")
-- OpenMP include dirs:
-- OpenMP libraries:    /usr/local/apps/gcc/13.2.0/lib64/libgomp.so;/lib64/libpthread.a
-- Found CUDAToolkit: /usr/local/cuda-9.2/include (found version "9.2.88")
-- CUDA include dirs: /usr/local/cuda-9.2/include
-- CUDA driver library: /usr/local/cuda-9.2/lib64
-- Looking for DPC++
-- Could NOT find DPCPP (missing: SYCL_INCLUDE_DIRS SYCL_LIBRARIES)
-- Could NOT find HIP (missing: HIP_ROOT_DIR HIP_INCLUDE_DIRS HIP_LIBRARIES HIP_RUNTIME_DEFINE HIP_HIPCONFIG_EXECUTABLE HIP_PLATFORM HIP_COMPILER)
-- OCCA_CUDA_ENABLED: 1
-- OCCA_HIP_ENABLED: 0
-- OCCA_DPCPP_ENABLED: 0
-- OCCA_OPENCL_ENABLED:
-- Checking if build type is 'Coverage'
-- Performing Test _frecursiveFlag
-- Performing Test _frecursiveFlag - Success
-- Build tests: OFF
-- Reducing RELEASE optimization level to O2
-- Looking for Fortran NONE - found
-- Looking for Fortran INT_CPU_TIME - found
-- Looking for Fortran EXT_ETIME - not found
-- Looking for Fortran EXT_ETIME_ - not found
-- Looking for Fortran INT_ETIME - found
-- --> Will use second_INT_ETIME.f and dsecnd_INT_ETIME.f as timing function.
-- Build deprecated routines: OFF
-- Build single precision real: OFF
-- Build double precision real: ON
-- Build single precision complex: OFF
-- Build double precision complex: OFF
-- Using supplied NETLIB BLAS implementation
-- Using supplied NETLIB LAPACK implementation
-- Verifying Fortran/C Compiler Compatibility
-- Verifying Fortran/C Compiler Compatibility - Success
-- Build html documentation: OFF
-- Build man documentation: OFF
-- Checking for a supported Fortran compiler
-- Checking for a supported Fortran compiler - Found the GNU Fortran compiler
-- Performing Test COMPILER_C_SUPPORTS_MCMODEL_MEDIUM
-- Performing Test COMPILER_C_SUPPORTS_MCMODEL_MEDIUM - Success
-- Performing Test COMPILER_C_SUPPORTS_MCMODEL_FPIC
-- Performing Test COMPILER_C_SUPPORTS_MCMODEL_FPIC - Success
-- Performing Test COMPILER_Fortran_SUPPORTS_LARGE_DATA_THRES
-- Performing Test COMPILER_Fortran_SUPPORTS_LARGE_DATA_THRES - Success

----------------- Summary -----------------
Installation directory: /home/aburbas/.local/nekrs
C compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpicc
C++ compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpic++
Fortran compiler: /usr/local/apps/openmpi/5.0.3-gcc11.3.1/bin/mpif77
Default backend : CUDA
CPU backend compiler: /usr/local/apps/gcc/13.2.0/bin/g++ (flags: -w -O3 -g -march=native -mtune=native -ffast-math)
NVIDIA CUDA backend enabled (flags: -w -O3 -lineinfo --use_fast_math)
GPU aware MPI support: ON
-------------------------------------------
-- Configuring done
-- Generating done
-- Build files have been written to: /home/aburbas/nekRS/build

cmake --build ./build --target install -j8

**Reply by stgeke** [reply]
This should be fixed in `next`
